tm2/pkg/p2p/transport.go

```diff
@@ -75,7 +75,10 @@ func NewMultiplexTransport(
 	mConfig conn.MConnConfig,
 	logger *slog.Logger,
 ) *MultiplexTransport {
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
Worth noting that this context is absolutely useless, because the lifecycle of the Transport is bananas, and its usage irregular

Either way, fine with moving it top-level like this 👍